### ISSUE TITLE
salt-run cache.grains tgt='xxx' return all minions cache grains

### ISSUE THREAD

**Issue by wu0407** [saltstack/salt#39519]
### Description of Issue/Question
running the 'salt-run cache.grains tgt='iZuf611tnyst338ckzoggbZ' ' return the all minions cache grains
### Setup
```
[root@xxx conft]# salt-key  -l 'acc' 
Accepted Keys:
iZuf611tnyst338ckzoggbZ
iZuf61xbxj05ogfdslq8csZ
iZuf62yhzoowqbdf9ti0suZ
iZuf62yhzoowqbdf9ti0svZ
iZuf62yhzoowqbdf9ti0swZ
iZuf62yhzoowqbdl6wu6r0Z
iZuf658mmyi41kqpclfudaZ
iZuf66nhr33hwhju6077yzZ
iZuf66nhr33hwhju6077z0Z
iZuf68u3ntdjcweqjk2zgcZ
iZuf68u3ntdjcweqjk2zgdZ
iZuf68u3ntdjcweqjk2zgeZ
iZuf68u3ntdjcweqjk2zgfZ
iZuf68u3ntdjcweqjk2zggZ
iZuf6g1rl8c4lo6vplpo4wZ
iZuf6g1rl8c4lo6vplpo4xZ
iZuf6g1rl8c4lo6vplpo4yZ
iZuf6g1rl8c4lo6vplpo4zZ
iZuf6g1rl8c4lo6vplpo50Z
iZuf6g1rl8c4lo6vplpo51Z
iZuf6g1rl8c4lo6vplpo52Z
```

### Steps to Reproduce Issue
```
[root@xxx conft]# salt-run cache.grains tgt='iZuf611tnyst338ckzoggbZ' |less
/usr/lib64/python2.6/site-packages/Crypto/Util/randpool.py:40: RandomPool_DeprecationWarning: This application uses RandomPool, which is BROKEN in older releases.  See http://www.pycrypto.org/randpool-broken
  RandomPool_DeprecationWarning)
iZuf611tnyst338ckzoggbZ:
    ----------
    SSDs:
    biosreleasedate:
        04/01/2014
    biosversion:
        rel-1.7.5-0-ge51488c-20140602_164612-nilsson.home.kraxel.org
    cpu_flags:
        - fpu
        - vme
        - de
        - pse
        - tsc
        - msr
.....
iZuf6g1rl8c4lo6vplpo50Z:
    ----------
    SSDs:
    biosreleasedate:
        04/01/2014
    biosversion:
        rel-1.7.5-0-ge51488c-20140602_164612-nilsson.home.kraxel.org
    cpu_flags:
        - fpu
        - vme
        - de
        - pse
```

### Versions Report
```
Salt Version:
           Salt: 2016.11.2
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug 18 2016, 15:13:37)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: centos 6.5 Final
        machine: x86_64
        release: 2.6.32-573.22.1.el6.x86_64
         system: Linux
        version: CentOS 6.5 Final
 
```

**Comment by morganwillcock** [comment]
Is a duplicate of #38003 and should already be fixed.

**Comment by wu0407** [comment]
it is will fix in next release?

**Comment by Ch3LL** [comment]
Yes this will be included in 2016.11.3
